docs/messaging/messaging-metrics.md

```diff
@@ -7,14 +7,12 @@
 <!-- toc -->
 
 - [Common attributes](#common-attributes)
+- [Metric: `messaging.client.operation.duration`](#metric-messagingclientoperationduration)
 - [Producer metrics](#producer-metrics)
-  - [Metric: `messaging.publish.duration`](#metric-messagingpublishduration)
-  - [Metric: `messaging.publish.messages`](#metric-messagingpublishmessages)
+  - [Metric: `messaging.client.published.messages`](#metric-messagingclientpublishedmessages)
 - [Consumer metrics](#consumer-metrics)
-  - [Metric: `messaging.receive.duration`](#metric-messagingreceiveduration)
-  - [Metric: `messaging.receive.messages`](#metric-messagingreceivemessages)
-  - [Metric: `messaging.process.duration`](#metric-messagingprocessduration)
-  - [Metric: `messaging.process.messages`](#metric-messagingprocessmessages)
+  - [Metric: `messaging.client.consumed.messages`](#metric-messagingclientconsumedmessages)
```

Comment:
If have some thoughts about the counters `messaging.client.[published|consumed].messages`.

1. I wonder if we should change those from opt-in and required for batching scenarios to always required? As now we don't anymore have a system-independent way of determining those values (we'd need to slice the `operation.duration` metrics by the exact and system specific operation name).
2. The pairing of the terms "publish" and "consume" seems a little bit odd to me, usually it's "produce" and "consume". Maybe "published" and "received" would be better?

Reply:
> I wonder if we should change those from opt-in and required for batching scenarios to always required? As now we don't anymore have a system-independent way of determining those values (we'd need to slice the operation.duration metrics by the exact and system specific operation name).

Agreed, done

Reply:
>The pairing of the terms "publish" and "consume" seems a little bit odd to me, usually it's "produce" and "consume". Maybe "published" and "received" would be better?

I really like `consumed` because it describes both receiving and processing. 
If I search for "publish and consume" I still see them used together, but I'm fine changing to `messaging.client.produced.messages`